src/main/java/com/odysseusinc/arachne/datanode/service/impl/AtlasServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class AtlasServiceImpl implements AtlasService {
+    private final RestTemplate atlasRestTemplate;
+    private final GenericConversionService conversionService;
+    private final SystemSettingsGroupRepository systemSettingsGroupRepository;
+    private HttpHeaders headers;
+
+    @Autowired
+    public AtlasServiceImpl(GenericConversionService genericConversionService,
+                            SystemSettingsGroupRepository settingsGroupRepository,
+                            @Qualifier("atlasRestTemplate") RestTemplate atlasRestTemplate) {
+
+        this.conversionService = genericConversionService;
+        this.systemSettingsGroupRepository = settingsGroupRepository;
+        this.atlasRestTemplate = atlasRestTemplate;
+    }
+
+
+    @Override
+    public String checkConnection() {
+
+        SystemSettingsGroupDTO systemSettingsGroup =
+                conversionService.convert(systemSettingsGroupRepository.findByName("atlas"), SystemSettingsGroupDTO.class);
+        Map<String, String> settings = systemSettingsGroup.getFieldList().stream().collect(Collectors.toMap(SystemSettingDTO::getName,
+                SystemSettingDTO::getValue));
+        String url = settings.get("atlas.host") + ":" + settings.get("atlas.port")
+                + (StringUtils.isBlank(settings.get("atlas.urlContext")) ? ""
+                : "/" + settings.get("atlas.urlContext"));
+        headers = new HttpHeaders();
+        headers.setContentType(MediaType.valueOf(MediaType.APPLICATION_FORM_URLENCODED_VALUE));
+
+        String atlasVersion = checkVersion(url);
+        String atlasToken = authToAtlas(url, settings.get("atlas.auth.schema"),
+                settings.get("atlas.auth.username"), settings.get("atlas.auth.password"));
+        headers.add(AtlasAuthRequestInterceptor.AUTHORIZATION_HEADER, AtlasAuthRequestInterceptor.BEARER_PREFIX + atlasToken);
+
+        getCohortDefinitionCount(url);
+
+        return Optional.ofNullable(atlasVersion).orElseThrow(
+                () -> new ServiceNotAvailableException("Atlas version is null"));
+
+    }
+
+    private String checkVersion(String url) {
+
+        AtlasClient.Info info;
+        try {
+            info = atlasRestTemplate.getForObject(
+                    new URI(url + Constants.Atlas.INFO), AtlasClient.Info.class);
+        } catch (Exception e) {
+            throw new ServiceNotAvailableException("Incorrect Atlas address");
+        }
+        return info != null ? info.version : null;
+    }
+
+    private String authToAtlas(String url, String authSchema, String login, String password) {
+
+        String atlasToken = null;
```

Comment:
I would suggest to re-use AtlasLoginClient:
1) Move contents of AtlasLoginClient Bean to parameterized method
2) Use that method to build the client
3) Use methods of that client